### PR TITLE
Fix #3592: [RTL] High-fi Add support for RTL in audio player and other drawables included in Oppia

### LINKED ISSUE #3592: [RTL] High-fi Add support for RTL in audio player and other drawables included in Oppia
- Audio Player in Exploration needs RTL support
- Other drwables used in the project should also need RTL support

screenshot

### PR DESCRIPTION
**Pull request by veena14cs**
## Explanation
This PR fixes #3592 . This PR fixes drawables used in Audio Player and also other drawables used in the project.

## Screenshot LTR and RTL

**Audio Player**

screenshot..... Screenshot_1627385683

screenshot.... Screenshot_1627385690

**Profile exit icon in Navigation Drawer**

Screenshot_1627642888.......Screenshot_1627642209

**Note : This screenshot only focus on audio player drawbles and not the image in exploration player.**

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by anandwana001 — CHANGES_REQUESTED** [review]
LGTM, before merging it, can we add a screenshot of all the screens contacting all the images which are updated here in PR. 

Also, should we add test for these?

**Comment by veena14cs** [comment]
> LGTM, before merging it, can we add a screenshot of all the screens contacting all the images which are updated here in PR.
> 
> Also, should we add test for these?

I m not sure of testcases. And regarding screenshots there are too many files actually to show LTR and RTL.

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_audio_lang_24px.xml]: For this I think the bubble should be mirrored but the internal text should not be. We might need a new icon for this.

@BenHenning WDYT?

File: app/src/main/res/drawable/ic_audio_streaming_off_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_audio_streaming_off_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_check_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_check_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_check_circle_primary_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_check_circle_primary_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_check_primary.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_check_primary.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_edit.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_edit.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_edit_18.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_edit_18.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_folder_open_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_folder_open_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_hide_eye_icon.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_hide_eye_icon.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_help_grey_icon_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_help_grey_icon_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_pause_circle_filled_white_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_pause_circle_filled_white_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_play_circle_filled_white_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_play_circle_filled_white_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/ic_upload_photo.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_upload_photo.xml]: This should not be reversed. Only icons which represent direction should be reversed.

File: app/src/main/res/drawable/icon_search_white_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/icon_search_white_24dp.xml]: This should not be reversed. Only icons which represent direction should be reversed.

Also reformat this file.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs PTAL.

File: app/src/main/res/drawable/icon_search_white_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/icon_search_white_24dp.xml]: Reverted and reformatted.

File: app/src/main/res/drawable/ic_upload_photo.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_upload_photo.xml]: Done.

File: app/src/main/res/drawable/ic_play_circle_filled_white_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_play_circle_filled_white_24dp.xml]: Done,

File: app/src/main/res/drawable/ic_pause_circle_filled_white_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_pause_circle_filled_white_24dp.xml]: Done,

File: app/src/main/res/drawable/ic_help_grey_icon_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_help_grey_icon_24dp.xml]: Reverted.

File: app/src/main/res/drawable/ic_check_primary.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_check_primary.xml]: Done.

File: app/src/main/res/drawable/ic_edit.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_edit.xml]: Done.

File: app/src/main/res/drawable/ic_check_circle_primary_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_check_circle_primary_24dp.xml]: Done,

File: app/src/main/res/drawable/ic_check_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_check_24dp.xml]: Done.

File: app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml]: Are you sure that we need not reverse this?

File: app/src/main/res/drawable/ic_audio_streaming_off_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_audio_streaming_off_24dp.xml]: Done.

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: @BenHenning Please confirm.

File: app/src/main/res/drawable/ic_folder_open_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_folder_open_24dp.xml]: Done.

File: app/src/main/res/drawable/ic_hide_eye_icon.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_hide_eye_icon.xml]: Done,

**Comment by veena14cs** [comment]
> @veena14cs PTAL.

Thanks @rt4914 I have addressed your comments PTAL.

File: app/src/main/res/drawable/ic_edit_18.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_edit_18.xml]: Done.

**Comment by veena14cs** [comment]
> LGTM, before merging it, can we add a screenshot of all the screens contacting all the images which are updated here in PR.
> 
> Also, should we add test for these?

Added screenshot.

**Review by anandwana001 — APPROVED** [review]
Overall LGTM

**Comment by veena14cs** [comment]
> Overall LGTM

Thanks.

File: app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml
Comment (rt4914) [reply app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml]: I have checked it by following these steps:
1. Try to find any app where we can see this icon getting used.
2. Take screenshot in LTR
3. Using `Developer Options`, change to RTL forcefully.
4. Now take screenshot of same previous screen in RTL.
5. Compare these screenshots for the icon to decide if mirroring is required or not.


Because of this I have mentioned that this should not be reversed.

**Comment by rt4914** [comment]
@veena14cs Replied to open comments.

File: app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_audio_streaming_on_24dp.xml]: Okay got it. Thanks for this info.

**Comment by veena14cs** [comment]
> @veena14cs Replied to open comments.

I have commented back please check.

**Review by rt4914 — APPROVED** [review]
Approved but needs reply on [this](https://github.com/oppia/oppia-android/pull/3593/files#r679424486) before merging.

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (BenHenning) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: Per https://material.io/design/usability/bidirectionality.html#mirroring-elements under 'Passage of Time' is this PR actually correct @rt4914? We shouldn't be mirroring progress bars or player buttons if they represent time since time is conventionally not mirrored in RTL contexts (unless I'm misunderstanding the bidi advice from that page).

**Review by BenHenning — DISMISSED** [review]

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (rt4914) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: @BenHenning We have not mirrored player buttons. But this icon is `language-change-icon` which I think might need some changes basically the background needs to be flipped. Also we have mirrored only 5 icons mentioned here: https://github.com/oppia/oppia-android/pull/3593/files

File: app/src/main/res/drawable/ic_refresh_white_24dp.xml
Comment (rt4914) [inline app/src/main/res/drawable/ic_refresh_white_24dp.xml]: This needs to be revert.

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (rt4914) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: @MohamedMedhat1998 Can you help us out in deciding what should be RTL behaviour of this icon:

Screenshot 2021-08-04 at 10 38 37 PM

1. If we do not mirror it it will look exactly same in RTL too.
2. If we mirror it fully than the text will be mirrored too which means that the character in foreign language will also get mirrored meaning it won't be a valid character anymore. So definitely we cannot do this.
Screenshot 2021-08-04 at 10 41 16 PM
3. Another option we have is to mirror the bubble/background only and keep the text exactly same for RTL.


What are your thoughts? What would be the best solution here?

**Review by rt4914 — CHANGES_REQUESTED** [review]
@veena14cs PTAL

File: app/src/main/res/drawable/ic_refresh_white_24dp.xml
Comment (veena14cs) [reply app/src/main/res/drawable/ic_refresh_white_24dp.xml]: Done.

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (BenHenning) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: Thanks @rt4914 for pointing out that I was mistaking this icon. Clearing my review for now since my codeowner approval isn't needed, and per our discussion this morning.

**Review by BenHenning — COMMENTED** [review]
Clearing my review since the approach overall looks correct, we just need to find a solution for the audio language icon.

**Comment by veena14cs** [comment]
> Clearing my review since the approach overall looks correct, we just need to find a solution for the audio language icon.

Thanks Ben!

File: app/src/main/res/drawable/ic_audio_lang_24px.xml
Comment (MohamedMedhat1998) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: > @MohamedMedhat1998 Can you help us out in deciding what should be RTL behaviour of this icon:
> 
> Screenshot 2021-08-04 at 10 38 37 PM
> 
> 1. If we do not mirror it it will look exactly same in RTL too.
> 2. If we mirror it fully than the text will be mirrored too which means that the character in foreign language will also get mirrored meaning it won't be a valid character anymore. So definitely we cannot do this.
> 
> Screenshot 2021-08-04 at 10 41 16 PM
> 
> 3. Another option we have is to mirror the bubble/background only and keep the text exactly same for RTL.
> What are your thoughts? What would be the best solution here?

Hi everyone. Thanks for mentioning me :)    
I think I haven't seen any mirrored icons throughout my life, all the icons remain the same but only their position changes.  
This doesn't apply to the navigation icons because they are always mirrored.  
.
I have checked the YouTube app and all the icons are the same in both locales. I have also checked Chome app and found only one mirrored icon, which is the "share" icon, but it looked weird to me btw. Also, the cellular signal strength icon found in the action bar of the device doesn't change.   
So to summarize my opinion  
1. Generally, don't mirror the icons.
2. Mirror only the navigation icons.  


According to this case specifically, if the player remains the same (which is the correct option according to what @BenHenning  has shared and to my experience with the players) then leave the icon as is. (don't mirror the text nor the background)

Comment (rt4914) [reply app/src/main/res/drawable/ic_audio_lang_24px.xml]: @veena14cs As per @MohamedMedhat1998 's comment can you undo this change.

**Review by rt4914 — APPROVED** [review]
@veena14cs I am unblocking you so that this PR can be merged but just undo the change in `ic_audio_lang_24px` and after which this PR can be merged.

**Comment by veena14cs** [comment]
> @veena14cs I am unblocking you so that this PR can be merged but just undo the change in `ic_audio_lang_24px` and after which this PR can be merged.

Thanks @rt4914 reverted changes done in `ic_audio_lang_24px`.

**Pull request merged**
